# Wilcoxon İşaretli Sıralar Testi (Wilcoxon Signed-Rank Test)
Wilcoxon Testi, Bağımlı Örneklem T Testi'nin (Konu 61) normallik sağlanmadığındaki non-parametrik karşılığıdır. İşaret Testi'nin "gelişmiş" versiyonu olarak da düşünülebilir.

## İşaret Testinden Farkı — Kritik Nokta
İşaret Testi **sadece yöne** (+ veya -) bakıyordu, farkların büyüklüğünü tamamen göz ardı ediyordu (73.6 ile 6.0, ikisi de sadece "+" sayılıyordu). **Wilcoxon ise hem yöne HEM büyüklüğe (sıralamaya) bakar** bu yüzden İşaret Testi'nden çok daha güçlü bir testtir.

## Nasıl Çalışır?
1. Her gözlem çiftinin farkını hesaplarız (sonrası - öncesi)
2. Farkların **mutlak değerlerini sıralıyoruz**
3. Her sıralamaya, orijinal farkın işaretini (+/-) geri veriyoruz.
4. Pozitif sıralamaların toplamı ile negatif sıralamaların toplamını karşılaştırıyoruz.

**Sonuçların Yorumlanması:** Eğer pozitif farklar hem daha **çok sayıda** hem de **daha büyük** ise, pozitif sıralamaların toplamı negatiflerden çok daha büyük çıkar bu, gerçek ve tutarlı bir yön değişimine işaret eder.

## Ne Zaman Kullanılır?
- Öncesi-sonrası (aynı bireyler) karşılaştırmasında, normallik sağlanmadığında
- Bağımlı/eşleştirilmiş iki örneklem söz konusu olduğunda

## Hipotezler
- **H0:** Farkların medyanı sıfırdır (öncesi-sonrası arasında sistematik bir fark yoktur)
- **H1:** Farkların medyanı sıfırdan farklıdır

## Python'da Kullanımı
```python
from scipy.stats import wilcoxon

istatistik, p_degeri = wilcoxon(oncesi, sonrasi, alternative='two-sided')
```
veya doğrudan farkları vererek:
```python
istatistik, p_degeri = wilcoxon(farklar)
```

In [1]:
import numpy as np
from scipy import stats as st

oncesi = np.random.normal(300, 50, 20)
sonrasi = oncesi + np.random.normal(15, 40, 20)

# H0: Farkların medyanı sıfıra eşittir, farklar simetrik dağılmıştır.
# H1: Farkların medyanı sıfırdan farklıdır.
farklar = sonrasi - oncesi
istatistik, p_degeri = st.wilcoxon(
    x=farklar,
    alternative='two-sided'
)
alpha = 0.05
print(f'Wilcoxon-T istatistiği: {istatistik}')
print(f'Wilcoxon-T P değeri: {p_degeri}')
if p_degeri < alpha: # type: ignore
    print('Farkların medyanı sıfırdan farklı olabilir.')
else:
    print('Farkların medyanı sıfıra eşittir, farklar simetrik dağılmıştır.')

Wilcoxon-T istatistiği: 22.0
Wilcoxon-T P değeri: 0.0010166168212890625
Farkların medyanı sıfırdan farklı olabilir.


In [2]:
print(f'Kampanya öncesi harcamaların ortalaması: {np.mean(oncesi)}')
print(f'Kampanya sonrası harcamaların ortalaması: {np.mean(sonrasi)}')
print(f'Kampanya öncesi harcamaların medyanı: {np.median(oncesi)}')
print(f'Kampanya sonrası harcamaların medyanı: {np.median(sonrasi)}')
print(f'Kampanya sonrası - öncesi harcamaların ortalaması: {np.mean(farklar)}')
print(f'Kampanya sonrası - öncesi harcamaların medyanı: {np.median(farklar)}')

Kampanya öncesi harcamaların ortalaması: 308.17818857322607
Kampanya sonrası harcamaların ortalaması: 326.8704590369408
Kampanya öncesi harcamaların medyanı: 285.615528029241
Kampanya sonrası harcamaların medyanı: 309.6243298535982
Kampanya sonrası - öncesi harcamaların ortalaması: 18.692270463714816
Kampanya sonrası - öncesi harcamaların medyanı: 19.92968310712456


### Sonuç
Kampanya öncesi ve sonrası harcamalar arasındaki farkın medyanının 
sıfırdan farklı olup olmadığını test etmek amacıyla Wilcoxon İşaretli 
Sıralar Testi uyguladık. Test sonuçlarına göre, farkların medyanı 
istatistiksel olarak anlamlı şekilde sıfırdan farklı bulunmuştur 
(W=22.0, p=0.001). Kampanya sonrası harcamaların hem ortalaması 
(326.87) hem medyanı (309.62), kampanya öncesine (sırasıyla 308.18 ve 
285.62) göre yüksektir; ortalama fark yaklaşık 18.7, medyan fark ise 
yaklaşık 19.9'dur. Bu doğrultuda, kampanyanın harcamaları anlamlı 
şekilde artırdığı söylenebilir.